bundles/framework/layerselector2/model/LayerGroup.class.js

```diff
@@ -1,12 +1,34 @@
 export class LayerGroup {
-    constructor (id, groupMethod, title) {
+    constructor (id, groupMethod, title, rootId, parentId, groups) {
         this.id = id;
         this.groupMethod = groupMethod;
         this.name = title;
         this.layers = [];
+        this.rootId = rootId || null;
+        this.parentId = parentId || null;
+        this.groups = groups || [];
         this.searchIndex = {};
         this.tools = [];
     }
+
+    getRootId () {
+        return this.rootId;
+    }
+    setRootId (rootId) {
+        this.rootId = rootId;
+    }
```

Comment:
Looks like there is no code that calls set or get for rootId. Is there a reason for these?